cpp/inc/bond/protocol/detail/rapidjson_helper.h

```diff
@@ -63,24 +62,53 @@ class RapidJsonInputStream
 public:
     typedef char Ch;
 
-    RapidJsonInputStream(typename boost::call_traits<Buffer>::reference input)
-        : input(&input),
+    explicit RapidJsonInputStream(const Buffer& buf)
+        : input(buf),
           current(0),
           count(0)
     {
         input.Read(current);
     }
 
-    RapidJsonInputStream(const RapidJsonInputStream& that, typename boost::call_traits<Buffer>::reference input)
-        : input(&input),
-          current(that.current),
-          count(that.count)
-    {}
+    RapidJsonInputStream(const RapidJsonInputStream&) = default;
+    RapidJsonInputStream& operator=(const RapidJsonInputStream&) = default;
+
+    #if defined(_MSC_VER) && _MSC_VER < 1900
+    // MSVC cannot = default rvalue ctor or move-assign operators
+    RapidJsonInputStream(RapidJsonInputStream&& other)
+        : input(std::move(other.input)),
+          current(std::move(other.current)),
+          count(std::move(other.count))
+    { }
+
+    RapidJsonInputStream& operator=(RapidJsonInputStream&& other)
+    {
+        input = std::move(other.input);
+        current = std::move(other.current);
+        count = std::move(other.count);
+        return *this;
+    }
+    #else
+    RapidJsonInputStream(RapidJsonInputStream&&) = default;
+    RapidJsonInputStream& operator=(RapidJsonInputStream&&) = default;
```

Comment:
I think you can simply skip all 4 `=default` copy/move constructors for non VC12 case.